test/Nancy.Tests/Unit/HttpLinkHeaderFixture.cs

```diff
@@ -0,0 +1,160 @@
+namespace Nancy.Tests.Unit
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Xunit;
+
+    public class HttpLinkHeaderFixture
+    {
+        private const string LinkHeader = "<http://nancyfx.org/>; "
+            + "rel=\"http://nancyfx.org/rel/Nancy\"; "
+            + "title=\"Nancy\", "
+            + "</alternate.jsonld>; "
+            + "rel=\"alternate\"; "
+            + "title=\"JSON LD Representation\"";
+
+        [Fact]
+        public void Add_existing_relation_in_different_casing_should_overwrite_the_relation()
+        {
+            // Given
+            var linkHeader = new HttpLinkHeader();
+
+            // When
+            linkHeader.Add(new HttpLink("/first", "ALTERNATE"));
+            linkHeader.Add(new HttpLink("/first", "AlterNate"));
+            linkHeader.Add(new HttpLink("/first", "alternate"));
+            linkHeader.Add(new HttpLink("/first", HttpLinkRelation.IanaLinkRelationPrefix + "alternate"));
+            // TODO: Figure out what to do with duplicates as per https://github.com/mnot/I-D/issues/170. @asbjornu

```

Comment:
@thecodejunkie I'm still discussing with @mnot in mnot/I-D#170. Once that's resolved and there's no more comments from the Custodians of the Super-Duper-Happy-Path, I'll squash (and rebase, if necessary) and force-push.